### PR TITLE
Add api to create ThreadFactory

### PR DESCRIPTION
**Pull request by JordiMartinezVicent**
Regarding the issue #3 I have creating this PR which adds the java 21 api to create Virtual Threads ThreadFactories. 

For example, on java 21 it would be: 

```java
//VirtualThread factory
ThreadFactory virtualThreadFactory = Thread.ofVirtual()
        .name("name")
        .uncaughtExceptionHandler(ueh)
        .factory();

```

With this api: 

```java
//VirtualThread factory
 ThreadFactory virtualThreadFactory = ThreadFactoryTool.ofVirtual()
        .name("name")
        .uncaughtExceptionHandler(ueh)
        .factory();
```

In order to add the ThreadFactory with the current code: 

```java

ThreadFactory factory = ThreadFactoryTool.ofVirtual()
  .name("name-", 1)
   factory();

ExecutorService executor = ExecutorTool.newThreadPerTaskExecutor(factory);
  executor.submit(() -> {
    System.out.println(Thread.currentThread().getName()); // prints "name-1"
});
```

### CODE REVIEW

**Comment by JordiMartinezVicent** [comment]
@maxxedev 

If you think is worthy to have the API, we can work on the PR to add it to the project.

File: src/main/java/com/github/thunkware/ThreadFactoryBuilderProvider.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+   * Creates a new {@link VirtualThreadFactoryBuilder}
+   * 
+   * @return new instance of VirtualThreadFactoryBuilder
+   */
+  VirtualThreadFactoryBuilder ofVirtual();
+
+  /**
+   * Builder to create Virtual Threads Factories
+   */
+  interface VirtualThreadFactoryBuilder {
```
Comment (maxxedev) [inline src/main/java/com/github/thunkware/ThreadFactoryBuilderProvider.java:21]: Java 21 defines the virtual thread builder as [Thread.Builder.OfVirtual](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/lang/Thread.Builder.OfVirtual.html)

I think we should define this similarly as ThreadTool.Builder.OfVirtual. What do you think?

**Comment by maxxedev** [comment]
@JordiMartinezVicent can you take a look at PR https://github.com/thunkware/virtual-threads-backport/pull/6

**Comment by JordiMartinezVicent** [comment]
> @JordiMartinezVicent can you take a look at PR #6

Sure! I like it. 

A part from the new api to create ThreadFactories I like the avoiding the reflection. 

I close this PR in favor of #6

**Pull request closed without merging**